pageserver/src/tenant/blob_io.rs

```diff
@@ -389,6 +389,21 @@ impl<const BUFFERED: bool> BlobWriter<BUFFERED> {
         };
         (srcbuf, res.map(|_| (offset, compression_info)))
     }
+
+    /// Writes a raw blob from raw header and data.
+    pub(crate) async fn write_blob_raw<Buf: IoBuf + Send>(
+        &mut self,
+        header: FullSlice<Buf>,
+        data: FullSlice<Buf>,
+        ctx: &RequestContext,
+    ) -> (FullSlice<Buf>, Result<u64, Error>) {
+        let offset = self.offset;
+        if let (_, Err(err)) = self.write_all(header, ctx).await {
+            return (data, Err(err));
+        }
+        let (data, res) = self.write_all(data, ctx).await;
+        (data, res.map(|_| offset))
+    }
```

Comment:
Why not use `write_blob_maybe_compressed(.., ImageCompressionAlgorithm::disabled)` instead?

Reply:
But the data is already compressed, we're just passing through the header and compressed data. Otherwise we'd set an uncompressed header for compressed data.